### PR TITLE
Add tvm::support::hexdump() debug utility

### PR DESCRIPTION
**Pull request by areusch**
Adding a debug utility which can be useful to inspect binary data in the C++ runtime.

As part of this change, build C++ tests against a statically-linked TVM runtime so that `-fvisibility=hidden` can still be used but doesn't apply to unit tests.

### CODE REVIEW

File: src/support/hexdump.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#define TVM_SUPPORT_HEXDUMP_H_
+
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+namespace tvm {
+namespace support {
+
+void hexdump(const std::string& s, std::ostream& os);
```
Comment (tqchen) [inline src/support/hexdump.h:35]: HexDump as per Google C convention

Comment (tqchen) [inline src/support/hexdump.h:35]: document via doxygen

Comment (areusch) [reply src/support/hexdump.h]: done

Comment (areusch) [reply src/support/hexdump.h]: done

**Comment by tqchen** [comment]
Thanks @areusch !

**Pull request merged**